src/System.Private.CoreLib/shared/System/IO/MemoryStream.cs

```diff
@@ -245,6 +245,25 @@ internal ReadOnlySpan<byte> InternalReadSpan(int count)
             return span;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal ReadOnlyMemory<byte> InternalReadMemory(int count)
```

Comment:
Is there benefit in having this piece of logic in a separate method? Why not just inline it into the one place where it is called?

Reply:
As the `InternalReadSpan` method is also used by the `BinaryReader` class, I assumed that the `InternalReadMemory` one could be used elsewhere. And also for consistency with sync-based code.

Reply:
You really only these two lines from the `InternalReadSpan/Memory`:

```
var span = new ReadOnlySpan<byte>(_buffer, _position, _length - _position);
_position = _length;
```

The rest is unnecessary for what you are using it for here. It would be better to just write these two lines that do exactly what you need and nothing else - for both the sync and async cases.

Reply:
OK. Thanks for your suggestion. It remains consistent and becomes more performant!